models/spring-ai-bedrock-converse/src/main/java/org/springframework/ai/bedrock/converse/BedrockProxyChatModel.java

```diff
@@ -448,8 +452,19 @@ private ChatResponse toChatResponse(ConverseResponse response) {
 			allGenerations.add(toolCallGeneration);
 		}
 
-		DefaultUsage usage = new DefaultUsage(response.usage().inputTokens().longValue(),
-				response.usage().outputTokens().longValue(), response.usage().totalTokens().longValue());
+		Long promptTokens = response.usage().inputTokens().longValue();
+		Long generationTokens = response.usage().outputTokens().longValue();
+		Long totalTokens = response.usage().totalTokens().longValue();
+
+		if (perviousChatResponse != null && perviousChatResponse.getMetadata() != null
+				&& perviousChatResponse.getMetadata().getUsage() != null) {
+
+			promptTokens += perviousChatResponse.getMetadata().getUsage().getPromptTokens();
```

Comment:
I have manually verified this step. Is there an effective way we can add a test to validate this additive operation in our tests?